ip/event_manager.go

```diff
@@ -131,7 +137,8 @@ func (m *manager) waitTwiceLeaseTimeOrReallocation(ctx context.Context) {
 // otherwise. It can happen if the current manager stopping has been cancelled.
 func (m *manager) stopOrder(ctx context.Context) bool {
 	m.stopMutex.RLock()
-	defer m.stopMutex.RUnlock()
+	stopper := m.stopper
```

Comment:
Are you sure the stopper function is _thread_ safe? Maybe we should say it in the comment of the stopper definition